File: DSA_Level-1/Recursion and Backtracking/Recursion with Arraylists/Get Stair Paths/GetStairPaths_Solution.java

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn=new Scanner(System.in);
        int n=scn.nextInt();
        ArrayList<String> res=getStairPaths(n);
        System.out.println(res);
    }

    public static ArrayList<String> getStairPaths(int n) {
        
        if(n==0){
            ArrayList<String> res=new ArrayList<>();
            res.add("");
            return res;
        }
        if(n<0){
            ArrayList<String> res=new ArrayList<>();
            return res;
        }
        
        ArrayList<String> path1=getStairPaths(n-1);
        ArrayList<String> path2=getStairPaths(n-2);
        ArrayList<String> path3=getStairPaths(n-3);
        
        ArrayList<String> paths=new ArrayList<>();
        
        for(String path:path1){
            paths.add("1"+path);
        }
        
        for(String path:path2){
            paths.add("2"+path);
        }
        
        for(String path:path3){
            paths.add("3"+path);
        }
        
        return paths;
    }

}

File: DSA_Level-1/Recursion and Backtracking/Recursion with Arraylists/Get Stair Paths/GetStairPaths_Solution.js

function getPaths(n){

    if(n==0){
        let res=[];
        res.push("");
        return res;
    }

    if(n<0){
        let res=[];
        return res;
    }

    let path1=getPaths(n-1);
    let path2=getPaths(n-2);
    let path3=getPaths(n-3);

    let paths=[];

    for(let s of path1){
        paths.push("1"+s);
    }

    for(let s of path2){
        paths.push("2"+s);
    }

    for(let s of path3){
        paths.push("3"+s);
    }

    return paths
}

console.log(getPaths(3));